tests/data-fetch-mixin.js

```diff
@@ -197,8 +197,91 @@ describe("DataFetch mixin", function() {
     var prevCallCount = fakeComponent.setState.callCount;
 
     var onError = $.ajax.args[0][0].error;
-    onError(null, 503, 'foobar');
+    onError({}, 503, 'foobar');
 
     expect(fakeComponent.setState.callCount).to.equal(prevCallCount);
-  })
+  });
+
+  it("should set isFetchingData to false in initial state", function() {
+    var initialState = fakeComponent.getInitialState();
+
+    expect(initialState.isFetchingData).to.equal(false);
+  });
+
+  it("should set dataError to null in initial state", function() {
+    var initialState = fakeComponent.getInitialState();
+
+    expect(initialState.dataError).to.equal(null);
+  });
+
+  describe("dataError set in this.state for failed requests", function() {
+    var url = 'www.foo.bar',
+        setStateArgs,
+        err = {toString: sinon.stub()},
+        xhrObj = {
+          status: 404,
+          statusText: 'Not found'
+        };
+
+    beforeEach(function() {
+      fakeComponent.props.dataUrl = url;
+
+      fakeComponent.componentWillMount();
+
+      var onError = $.ajax.args[0][0].error;
+      onError(xhrObj, xhrObj.status, err);
+
+      setStateArgs = fakeComponent.setState.lastCall.args[0];
+    });
+
+    afterEach(function() {
+      err.toString.reset();
+    });
+
+    it("should save the url of a failed request", function() {
+      expect(setStateArgs.dataError.url).to.equal(url);
+    });
+
+    it("should save the statusCode of a failed request", function() {
+      expect(setStateArgs.dataError.statusCode).to.equal(xhrObj.status);
+    });
+
+    it("should save the statusText of a failed request", function() {
+      expect(setStateArgs.dataError.statusText).to.equal(xhrObj.status);
+    });
+
+    it("should save the message of a failed request", function() {
+      expect(err.toString.callCount).to.equal(1);
```

Comment:
> You're not resetting the stub between tests

He [is](https://github.com/piatra/react-data-fetch/blob/fa2f8aacb5a218ec3acdaa373f54706aa41588b0/tests/data-fetch-mixin.js#L238)n't?

> we should put the entire error object in the state so the user can do whatever e wants with it

Maybe, maybe not. I'm reluctant when it comes to dumping a rich instance into the state, feels wasteful and even though it is technically serializable it doesn't feel so. Let's still to this for now and see what are needs are in the future.